### PR TITLE
Don’t show a “Preparing current file” progress while determining target of file

### PR DESCRIPTION
**Pull request by ahoppen**
Every time a request regarding a text document is sent, we call `schedulePreparationForEditorFunctionality` on the `SemanticIndexManager` with the assumption that it’s a no-op if the target is already prepared. We did, however, show the “Preparing current file” status while we were determining the file‘s target to determine that the file’s target is up-to-date.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Tests/SourceKitLSPTests/BackgroundIndexingTests.swift [hunk 1]
```diff
@@ -868,4 +868,23 @@ final class BackgroundIndexingTests: XCTestCase {
...
+    let project = try await SwiftPMTestProject(
+      files: [
+        "Lib.swift": ""
+      ],
+      capabilities: ClientCapabilities(window: WindowClientCapabilities(workDoneProgress: true)),
+      enableBackgroundIndexing: true
+    )
+
+    // Opening the document prepares it for editor functionality. Its target is already prepared, so we shouldn't show
+    // a work done progress for it.
```
Comment (bnbarham) [inline Tests/SourceKitLSPTests/BackgroundIndexingTests.swift:882]: What make sit already prepared?

**Review by bnbarham — APPROVED** [review]

File: Tests/SourceKitLSPTests/BackgroundIndexingTests.swift (hunk 1 above)
Comment (ahoppen) [reply Tests/SourceKitLSPTests/BackgroundIndexingTests.swift:882]: Creating a `SwiftPMTestProject` polls the index (`pollIndex` parameter in the initializer defaults to `true`), so just creating the project will ensure that the index is up-to-date.

**Review by hamishknight — COMMENTED** [review]
LGTM (given you added the `defer` in another PR)

**Pull request merged**